cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenCheckout.js

```diff
@@ -84,6 +83,41 @@ function renderGenericComponent() {
         var checkout = new AdyenCheckout(checkoutConfiguration);
         document.querySelector("#paymentMethodsList").innerHTML = "";
         var paymentMethodsUI = document.querySelector('#paymentMethodsList');
+
+        if(data.AdyenPaymentMethods.storedPaymentMethods) {
+            for (var i = 0; i < data.AdyenPaymentMethods.storedPaymentMethods.length; i++) {
+                var paymentMethod = data.AdyenPaymentMethods.storedPaymentMethods[i];
```

Comment:
We should add a filter on supportedShopperInteractions = Ecommerce in this loop. This will indicate the OneClick contracts. I assume that this now also renders Recurring contracts (supportedShopperInteractions = ContAuth), which we don't want to display on the checkout page